### ISSUE TITLE
Why does the Loss Function compare the model's output directly with the original data?

### ISSUE THREAD

**Issue by VRWpanda** [mobicom24/RF-Diffusion#25]
```
    def train_iter(self, features):
        self.optimizer.zero_grad()
        data = features['data']  # orignial data, x_0, [B, N, S*A, 2]
        cond = features['cond']  # cond, c, [B, C]
        B = data.shape[0]
        # random diffusion step, [B]
        t = torch.randint(0, self.diffusion.max_step, [B], dtype=torch.int64)
        degrade_data = self.diffusion.degrade_fn(
            data, t ,self.task_id)  # degrade data, x_t, [B, N, S*A, 2]
        predicted = self.model(degrade_data, t, cond)
        if self.task_id==3:
            data = data.reshape(-1,512,1,2)
        loss = self.loss_fn(data, predicted)  #TODO: change to noise loss
        loss.backward()
        self.grad_norm = nn.utils.clip_grad_norm_(
            self.model.parameters(), self.params.max_grad_norm or 1e9)
        self.optimizer.step()
        return loss
```

The current training loop compares the model's output directly with the original data (x_0), but based on standard diffusion model theory, the model should predict noise (ε) rather than the original data.
